### DISCUSSION TITLE
Configuration File Templates

### DISCUSSION THREAD

**Discussion by ghost** [Jyers/Marlin#900] [Q&A] [answered]
there seems to be a lot of differences between the configuration files in the marlin folder and the configuration file templates besides those related to the template. Is this supposed to be like that?

I compiled with the `Configuration Files\E3V2 Templates\UBL-BLTouch-10x10` files, and I have so far already ran into problems (i.e. heating failed) which according to the commits should have been solved.

I suspect that (some of) the changes applied to the configuration files in the marlin folder, were also meant to be applied to the templates.

**Comment by smooreace** [comment]
The files in the Marlin folder are the default settings from Marlin. From
what I've seen they won't even compile, and definitely won't work on a v2.
Ide has to check to be absolutely sure, but I think all of the pre-compiled
firmware versions that are available are based on compiles with the
same-named config file "set". So they should be compiling fine. You wont
get far in requesting much more specific help though without posting up
some errors to work from

**Comment by Reggi3** [comment]
Yes, it is supposed to be like that, the ones in the marlin folder are the original files marlin ships with.  The ones that are specific to us are in:
[the configuration files folder](https://github.com/Jyers/Marlin/tree/Ender_3_V2_Extensible_UI/Configuration%20Files)

They're vastly different because they're supposed to be.

**Comment by ghost** [comment]
@smooreace  @Reggi3
Just so there is no misunderstanding, I'm talking about these files:

[Ender_3_V2_Extensible_UI\Marlin\Configuration_adv.h](https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Marlin/Configuration_adv.h)
[Ender_3_V2_Extensible_UI\Marlin\Configuration.h](https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Marlin/Configuration.h)

and comparing them, in my case, to:

[Ender_3_V2_Extensible_UI\Configuration Files\E3V2 Templates\UBL-BLTouch-10x10\Configuration_adv.h](https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Configuration%20Files/E3V2%20Templates/UBL-BLTouch-10x10/Configuration_adv.h)
[Ender_3_V2_Extensible_UI\Configuration Files\E3V2 Templates\UBL-BLTouch-10x10\Configuration.h](https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Configuration%20Files/E3V2%20Templates/UBL-BLTouch-10x10/Configuration.h)

**Comment by ghost** [comment, marked as answer]
I'm going to just try too change the settings and see if that fixes my problems.

**Comment by smooreace** [comment]
As both of us have already stated...

Ender_3_V2_Extensible_UI\Marlin\Configuration_adv.h
<https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Marlin/Configuration_adv.h>
Ender_3_V2_Extensible_UI\Marlin\Configuration.h
<https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Marlin/Configuration.h>

are stock default Marlin config files that have literally NOTHING to do
with any printer specifically. They shouldn't work as they are for probably
any of the hundreds of printers the firmware is used on. That are not to be
used as-it

Thats why there are E3v2 template files provided, and yes, they are vastly
different because they ARE meant to be used on... and Ender 3 v2
specifically

**Reply by Reggi3** [reply]
No confusion this end, it's part of the marlin instructions on how to use, download the marlin sources, download the configuration files that match the branch of marlin that you downloaded, copy the configs for the printer that you're using to the marlin folder and change things as necessary, compile.

With Jyers branch, the config files that you would download separately and copy into the marlin folder are in the Configuration Files folder  :-)
